packages/contracts/contracts/ChugSplashRegistry.sol

```diff
@@ -1,50 +1,312 @@
+     * Emitted whenever there is a new manager for an existing project.
+     */
+    event ChugSplashManagerUpdated(
+        string indexed projectName,
+        address indexed previousManager,
+        address indexed newManager
+    );
+
+    /**
+     * Mapping of project names to ChugSplash projects.
+     */
+    mapping(string => ChugSplashProject) public projects;
+
+    /**
+     * Allows only the manager of a project to call the functions.
      */
-    mapping(string => ChugSplashManager) public registry;
+    modifier onlyManager(string memory _name) {
+        require(msg.sender == projects[_name].manager, "ChugSplashRegistry: caller is not manager");
+        _;
+    }
 
     /**
-     * Registers a new project and deploys a corresponding ChugSplashManager contract.
+     * Registers a new project.
      *
      * @param _name Name of the new ChugSplash project.
-     * @param _owner Initial owner for the new ChugSplashManager contract.
-     * @return Address of the newly deployed ChugSplashManager contract.
+     * @param _manager Initial manager for the new project.
+     * @return Address of the manager for the new project.
      */
-    function register(string memory _name, address _owner) public returns (address) {
+    function register(string memory _name, address _manager) public returns (address) {
         // TODO: Standardize error reporting system.
         require(
-            address(registry[_name]) == address(0),
+            projects[_name].manager == address(0),
             "ChugSplashRegistry: name already registered"
         );
 
-        // By making the salt be the hash of the name, we can easily predict the address of the
-        // ChugSplashManager contract based on the project's name.
-        bytes32 salt = keccak256(bytes(_name));
-        registry[_name] = new ChugSplashManager{ salt: salt }(_owner);
-        emit ChugSplashProjectRegistered(_name, msg.sender, _owner, address(registry[_name]));
-        return address(registry[_name]);
+        ChugSplashProject storage project = projects[_name];
+        project.manager = _manager;
+        emit ChugSplashProjectRegistered(_name, msg.sender, _manager);
+
+        // TODO: We used to return the ChugSplashManager address here. Should we keep this new
+        // return value or remove it?
+        return _manager;
+    }
+
+    /**
+     * Allows a manager to propose a new ChugSplash bundle to be executed.
+     *
+     * @param _name Name of the ChugSplash project.
+     * @param _bundleHash Hash of the bundle to execute.
+     * @param _bundleSize Total number of actions in the bundle.
+     * @param _configUri URI pointing to the config file for the bundle.
+     */
+    function proposeChugSplashBundle(
+        string memory _name,
+        bytes32 _bundleHash,
+        uint256 _bundleSize,
+        string memory _configUri
+    ) public onlyManager(_name) {
+        ChugSplashProject storage project = projects[_name];
```

Comment:
An important here is to have the ability to have multiple proposed bundles for the same project at the same time (but only one active bundle at a time).